### PR TITLE
nvptx-tools: add v2023-09-13

### PR DESCRIPTION
**Pull request by pranav-sivaraman**
Latest nvptx-tools has changes that address errors building gcc with later CUDA versions. 

Mostly likely fixes this [issue](https://github.com/spack/spack/issues/36172#issue-1628083572).

I was able to build `gcc@13.2.0 ^cuda@12.3.0` and had no issues compiling and running a test app.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the new version commit.

**Comment by wyphan** [comment]
I like this approach of spinning off `nvptx-tools` from a GCC `resource` into its own package. Maybe I should do the same for `newlib`? In any case, I'll rebase #35919 as soon as this PR gets merged, since I made some adjustments to the GCC NVPTX build part too.

**Comment by wyphan** [comment]
Also, once the `nvptx-tools` package is in place, you can now replace the `resource` block in the GCC package with this:
```
depends_on("nvptx-tools@2023-09-13", when="+nvptx")
```
And I think I know how to overcome this comment:
https://github.com/spack/spack/blob/a2f00886e911a8219bfac27752e5c7fd83c65280/var/spack/repos/builtin/packages/gcc/package.py#L310-L312

**Comment by wyphan** [comment]
@pranav-sivaraman I just successfully built and tested `gcc@13.2.0 +nvptx ^nvptx-tools@2023-09-13`. The fix is kinda ugly: copy all the contents of `nvptx-tools` install prefix into the `gcc` install prefix. But it works!
Since I can't push into your branch, here's the diff patch on top of your latest commit.
[gcc-nvptx-tools.patch](https://github.com/spack/spack/files/13619095/gcc-nvptx-tools.patch)

File: var/spack/repos/builtin/packages/gcc/package.py [hunk 1]
```diff
@@ -860,6 +857,28 @@ def configure_args(self):
 
         return options
 
+    # Copy nvptx-tools into the GCC install prefix
+    def copy_nvptx_tools(self):
+        nvptx_tools_bin_path = join_path(self.spec["nvptx-tools"].prefix, "bin")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/gcc/package.py]: Technically you don't need a `join_path` here.  You can instead:

```suggestion
        nvptx_tools_bin_path = self.spec["nvptx-tools"].prefix.bin
```

File: var/spack/repos/builtin/packages/gcc/package.py [hunk 2]
```diff
@@ -860,6 +857,28 @@ def configure_args(self):
...
+        gcc_bin_path = join_path(self.prefix, "bin")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/gcc/package.py]: Same here:

```suggestion
        gcc_bin_path = self.prefix.bin
```

**Review by tldahlgren — DISMISSED** [review]
Confirmed the new version commit.  Noted spec prefixes don't need to be `join_path`'d with their `bin` directories.  For more information, refer to the documentation on prefix objects at https://spack.readthedocs.io/en/latest/packaging_guide.html#prefix-objects.

**Comment by tldahlgren** [comment]
@alalazo @michaelkuhn Are you okay with the changes to the `gcc` package?

**Review by michaelkuhn — APPROVED** [review]
Sorry for being late to the party. The GCC changes LGTM and I just gave them a brief try.

**Pull request merged**